### PR TITLE
feat: support cdn url

### PR DESCRIPTION
**Pull request by l1ch40**
Resolve https://github.com/zineland/zine/issues/71

### CODE REVIEW

File: src/html.rs [hunk 1]
```diff
@@ -31,6 +31,45 @@ impl<'a> Meta<'a> {
     }
 }
 
+/// Rewrite root path URL of static files in `raw_html` with `cdn_url`.
+pub fn rewrite_html_cdn_url(raw_html: &[u8], cdn_url: &str) -> Result<Vec<u8>> {
```
Comment (Folyd) [inline src/html.rs:35]: I suggest merging `rewrite_cdn_url` and `rewrite_base_url` into a single function for better performace.

**Review by Folyd — CHANGES_REQUESTED** [review]
Thanks for working on this.

File: src/html.rs (hunk 1 above)
Comment (l1ch40) [reply src/html.rs]: Please, Review this commit https://github.com/zineland/zine/pull/140/commits/259eb5630f33d9aa82f4f573c9840ce7f15cce75

**Review by Folyd — CHANGES_REQUESTED** [review]
I mean we can rewrite `base_url` and `cdn_url` in a single rewriting, no need to call `rewrite_html_url` multiple times. 
```rust
pub fn rewrite_html_url(
    raw_html: &[u8],
    base_url: Option<&str>,
    cdn_url: Option<&str>,
) -> Result<Vec<u8>> {
}
```
Similar to this:
```rust
if matches!(current_mode(), Mode::Build) {
    rewrite_html_url(&buf, site.get("url"), site.get("cdn"))?;
}
```

**Comment by l1ch40** [comment]
> I mean we can rewrite `base_url` and `cdn_url` in a single rewriting, no need to call `rewrite_html_url` multiple times.
> 
> ```rust
> pub fn rewrite_html_url(
>     raw_html: &[u8],
>     base_url: Option<&str>,
>     cdn_url: Option<&str>,
> ) -> Result<Vec<u8>> {
> }
> ```
> 
> Similar to this:
> 
> ```rust
> if matches!(current_mode(), Mode::Build) {
>     rewrite_html_url(&buf, site.get("url"), site.get("cdn"))?;
> }
> ```

Get

**Review by Folyd — CHANGES_REQUESTED** [review]
LGTM overall. However, I believe we should remove `/static` part from the cdn URL. For example, `/static/test.png` should be `https://cdn-example.net/test.png`.

**Comment by l1ch40** [comment]
> LGTM overall. However, I believe we should remove `/static` part from the cdn URL. For example, `/static/test.png` should be `https://cdn-example.net/test.png`.

 Please,  Review the commit https://github.com/zineland/zine/pull/140/commits/ad67036c3b789cb45b5395ecd4f74d56a7e4cd3a

File: src/html.rs [hunk 2]
```diff
@@ -38,9 +38,10 @@ pub fn rewrite_html_base_url(
     cdn_url: Option<&str>,
 ) -> Result<Vec<u8>> {
     let rewrite_url_in_attr = |el: &mut Element, attr_name: &str| {
-        if let Some(attr) = el.get_attribute(attr_name) {
+        if let Some(mut attr) = el.get_attribute(attr_name) {
             let mut base_url = "";
             if attr.starts_with("/static") {
+                attr = attr.replacen("/static", "", 1);
```
Comment (Folyd) [inline src/html.rs:44]: We can use `strip_prefix()`:

```rust
if let Some(attr) = attr.strip_prefix("/static") {

}
```

Comment (l1ch40) [reply src/html.rs]: please https://github.com/zineland/zine/pull/140/commits/3672c321ba8576c47dbf488bb2abcf5e60a15c3b

**Pull request merged**